addons/account/models/account_lock_exception.py

```diff
@@ -277,52 +277,54 @@ def _get_active_exceptions_domain(self, company, soft_lock_date_fields):
     def _get_audit_trail_during_exception_domain(self):
         self.ensure_one()
 
-        domain = [
-            ('model', '=', 'account.move'),
-            ('account_audit_log_activated', '=', True),
-            ('message_type', '=', 'notification'),
-            ('account_audit_log_move_id.company_id', 'child_of', self.company_id.id),  # WORKAROUND: record_company_id is not set for bills
+        common_message_domain = [
             ('date', '>=', self.create_date),
         ]
-
         if self.user_id:
-            domain.append(('create_uid', '=', self.user_id.id))
+            common_message_domain.append(('create_uid', '=', self.user_id.id))
         if self.end_datetime:
-            domain.append(('date', '<=', self.end_datetime))
+            common_message_domain.append(('date', '<=', self.end_datetime))
 
-        # Add a restriction on the accounting date to avoid unnecessary entries
+        # Add restrictions on the accounting date to avoid unnecessary entries
         min_date = self.lock_date
         max_date = self.company_lock_date
         move_date_domain = []
         tracking_old_datetime_domain = []
         tracking_new_datetime_domain = []
         if min_date:
-            move_date_domain.append([('account_audit_log_move_id.date', '>=', min_date)])
+            move_date_domain.append([('date', '>=', min_date)])
             tracking_old_datetime_domain.append([('tracking_value_ids.old_value_datetime', '>=', min_date)])
             tracking_new_datetime_domain.append([('tracking_value_ids.new_value_datetime', '>=', min_date)])
         if max_date:
-            move_date_domain.append([('account_audit_log_move_id.date', '<=', max_date)])
+            move_date_domain.append([('date', '<=', max_date)])
             tracking_old_datetime_domain.append([('tracking_value_ids.old_value_datetime', '<=', max_date)])
             tracking_new_datetime_domain.append([('tracking_value_ids.new_value_datetime', '<=', max_date)])
-        domain.extend([
+
+        return [
+            ('company_id', 'child_of', self.company_id.id),
```

Comment:
I think it's better to filter more before the OR operator

```suggestion
            ('company_id', 'child_of', self.company_id.id),
            ('audit_trail_message_ids', 'any', common_message_domain),
```

Reply:
Makes sense, thanks